02b_threads_by_Marcin/pararel_count_if.cc

```diff
@@ -0,0 +1,122 @@
+#include <bits/stdc++.h>
+#include <iostream>
+#include <thread>
+#include <iterator>
+
+// #include <numeric>
+#include <algorithm>
+#include <vector>
+#include <chrono>
+// #include <functional>
+
+#include <bits/stdc++.h> 
+
+constexpr size_t minimumSize = 200;
+
+struct debugTimers
+{
+	decltype(std::chrono::steady_clock::now()) start3;
+	decltype(start3) stop3;
+	decltype(start3) start4;
+	decltype(start4) stop4;
+};
+
+// Check is numbe is even
+bool isEven(int i) 
+{ 
+    return (i % 2 == 0) ? true : false;
+} 
+
+debugTimers dummy_dt;
+
+template <typename IT, typename T>
+typename std::iterator_traits<IT>::difference_type
+pararelCountIf(IT first, IT last, T fun, size_t minSizeForThread = minimumSize, debugTimers& dt = dummy_dt ) {
+	const size_t size = std::distance(first, last);
+	if (size < minSizeForThread)
+		return std::count_if(first, last, fun);
+
+	size_t hardwareThread = std::thread::hardware_concurrency();
+
+    if (hardwareThread == 0)
+        hardwareThread = 1;
```

Comment:
jak jest 1 wątek to lepiej chyba zrobić standardowego std::count_if na całym zakresie. 
Potem będzie tworzenie pustych wektorów poniżej :)
Lepiej dać 2. 0 oznacza, że nie wiemy ile ich jest.